python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -811,51 +820,10 @@ def _init(self, *args, **kwds):
         for k in kwds:
             if stripProxy(self)._schema.hasAttribute(k):
                 this_arg = stripProxy(kwds[k])
-                if hasattr(this_arg, '_auto__init__'):
-                    this_arg._auto__init__()
-
-                ## Copying this from the __set__ method in the Proxy descriptor
-
-                if k == 'application':
-                    ProxyDataDescriptor.__app_set__(self, this_arg)
-                if k == 'is_prepared':
-                    ProxyDataDescriptor.__prep_set__(self, this_arg)
-
-
                 raw_self = stripProxy(self)
-
-                if isinstance(this_arg, str):
-                    this_arg = stripProxy(runtimeEvalString(raw_self, k, this_arg))
-                    if hasattr(this_arg, '_auto__init__'):
-                        this_arg._auto__init__()
-
-                if isinstance(this_arg, str):
-                    raw_self.setSchemaAttribute(k, raw_self._attribute_filter__set__(k, this_arg))
-                    continue
-                else:
-                    item = pluginclass._schema.getItem(k)
-
-                    # unwrap proxy
-                    if item.isA(ComponentItem):
-                        from .Filters import allComponentFilters
-                        cfilter = allComponentFilters[item['category']]
-                        stripper = lambda v: stripComponentObject(v, cfilter, item)
-                    else:
-                        stripper = None
-
-                    if item['sequence']:
-                        this_arg = ProxyDataDescriptor.__sequence_set__(stripper, raw_self, this_arg, k)
-                    else:
-                        if stripper is not None:
-                            this_arg = stripper(this_arg)
-                    # apply attribute filter to component items
-                    if item.isA(ComponentItem):
-                        this_arg = ProxyDataDescriptor._stripAttribute(raw_self, this_arg, k)
-
-                    if hasattr(this_arg, '_auto__init__'):
-                        this_arg._auto__init__()
-
-                    raw_self.setSchemaAttribute(k, raw_self._attribute_filter__set__(k, this_arg))
+                # This calls the same logic when assigning a named attribute as when we're assigning it to the object
+                # There is logic here which we 'could' duplicate but it is over 100 lines of code which then is duplicating funtionality written elsewhere
+                ProxyDataDescriptor(k, False).__set__(raw_self, raw_self._attribute_filter__set__(k, this_arg))
```

Comment:
I would have argued the cost of making this object is surprisingly cheap but if we're saved from that then fantastic. OK, feel free to merge it against this PR when ready if there aren't any problems